src/lib/models/ConfigModel.ts

```diff
@@ -10,48 +11,85 @@ export interface ConfigModel {
-#      'severity' - [Optional] The severity level value that you want to use to override the default severity level for the rule
-#                   Possible values: 1 or 'Critical', 2 or 'High', 3 or 'Moderate', 4 or 'Low', 5 or 'Info'
-#      'tags'     - [Optional] The string array of tag values that you want to use to override the default tags for the rule
-#
-# [Example usage]:
-# rules:
-#   eslint:
-#     sort-vars:
-#       severity: Info
-#       tags: ["Recommended", "Suggestion"]
-rules: {} # Remove this empty object {} when you are ready to specify your first rule override
-
-# Engine specific custom configuration settings of the format engines.<engine_name>.<property_name> = <value> where
+function yamlFormatConfig(rawConfig: CodeAnalyzerConfig, ruleSelection: RuleSelection): string {
+
+	let results = yamlFormatConfigProperty('config_root', rawConfig.getConfigRoot(), true)
+		+ '\n'
+		+ yamlFormatConfigProperty('log_folder', rawConfig.getLogFolder(), true)
+		+ '\n';
+	const customEngineModules: string[] = rawConfig.getCustomEnginePluginModules();
+	const customEngineModulesString: string = customEngineModules.length > 0
+		? `["${customEngineModules.join('", "')}"]`
+		: `[]`;
+	results += yamlFormatConfigProperty('custom_engine_plugin_modules', customEngineModulesString, false)
+		+ '\n'
+		+ yamlFormatRuleSelection(ruleSelection)
+		+ '\n'
+		+ DUMMY_ENGINE_SECTION
+		+ '\n';
+	return results;
+}
+
+function yamlFormatConfigProperty(propertyName: string, propertyValue: string, isCalculated: boolean): string {
+	// Get the annotation and prepend each line with '# ' to make it a YAML comment.
+	const annotation: string = getMessage(BundleName.ConfigModel, `annotation.${propertyName}`)
+		.replace(/^./gm, s => `# ${s}`);
+	const lastCalculatedAs: string = getMessage(BundleName.ConfigModel, 'template.last-calculated-as', [propertyValue]);
+
+	return `${annotation}\n`
+		+ `${propertyName}: ${propertyValue}${isCalculated ? ' # ' + lastCalculatedAs : ''}\n`;
+}
+
+function yamlFormatRuleSelection(ruleSelection: RuleSelection): string {
+	// Get the annotation and prepend each line with '# ' to make it a YAML comment.
+	const annotation: string = getMessage(BundleName.ConfigModel, `annotation.rules`)
+		.replace(/^./gm, s => `# ${s}`);
+
+	if (ruleSelection.getCount() === 0) {
+		return `${annotation}\n`
+			+ `rules: {} # Remove this empty object {} when you are ready to specify your first rule override`;
+	}
+
+	let results = annotation + '\n'
+		+ 'rules:\n';
+
+	for (const engine of ruleSelection.getEngineNames()) {
+		results += `  ${engine}:\n`;
+		for (const rule of ruleSelection.getRulesFor(engine)) {
+			const ruleName: string = rule.getName();
+			const severity = `${rule.getSeverityLevel()}`;
+			const tags = rule.getTags();
+			const tagString = tags.length > 0 ? `["${tags.join('", "')}"]` : `[]`;
+			// This section is built by hand instead of using a message, because the message catalog dislikes indentation.
```

Comment:
I don't think we need this comment. The message catalog is only used for translation purposes... i.e. we put things in there in english so that in the future we have a single place for translation. For just formatting strings... we wouldn't be using a message catalog.

Reply:
We typically use the message catalog to construct user-facing strings whenever possible, so I added this comment to explain why we're not doing that in this instance.
Technically, I could have made `severity` and `tags` strings in the catalog, but hat felt like overkill.

Reply:
Yeah, i'm just challenging that practice... I personally never worked on a team that put things like '%s. %s' in the message catalog. I think only things that need to be reviewed by documentation and things that need to be translated go into a message catalog. So I don't think the comment is needed if we change our assumptions here.